internal/groups/http.go

```diff
@@ -0,0 +1,685 @@
+			return groupPageRes{}, err
+		}
+		return buildGroupsResponse(gp), nil
+	}
+}
+
+func DeleteGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(groupReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		if err := svc.RemoveGroup(ctx, req.token, req.groupID); err != nil {
+			return nil, err
+		}
+		return groupDeleteRes{}, nil
+	}
+}
+
+func AssignMemberToGroup(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(memberGroupReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		if err := svc.Assign(ctx, req.token, req.memberID, req.groupID); err != nil {
+			return nil, err
+		}
+		return assignMemberToGroupRes{}, nil
+	}
+}
+
+func RemoveMemberFromGroup(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(memberGroupReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		if err := svc.Unassign(ctx, req.token, req.memberID, req.groupID); err != nil {
+			return nil, err
+		}
+		return removeMemberFromGroupRes{}, nil
+	}
+}
+
+func ListMembersForGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listMemberGroupReq)
+		if err := req.validate(); err != nil {
+			return memberPageRes{}, err
+		}
+		mp, err := svc.ListMembers(ctx, req.token, req.groupID, req.offset, req.limit, req.metadata)
+		if err != nil {
+			return memberPageRes{}, err
+		}
+		return buildUsersResponse(mp), nil
+	}
+}
+
+func UpdateGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(updateGroupReq)
+		if err := req.validate(); err != nil {
+			return createGroupRes{}, err
+		}
+
+		group := Group{
+			Description: req.Description,
+			ParentID:    req.ParentID,
+			Metadata:    req.Metadata,
+		}
+
+		g, err := svc.UpdateGroup(ctx, req.token, group)
+		if err != nil {
+			return createGroupRes{}, err
+		}
+
+		res := createGroupRes{
+			Name:        g.Name,
+			Description: g.Description,
+			Metadata:    g.Metadata,
+			created:     false,
+		}
+		return res, nil
+	}
+}
+
+func ViewGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(groupReq)
+		if err := req.validate(); err != nil {
+			return viewGroupRes{}, err
+		}
+
+		g, err := svc.ViewGroup(ctx, req.token, req.groupID)
+		if err != nil {
+			return viewGroupRes{}, err
+		}
+		res := viewGroupRes{
+			Name:        g.Name,
+			Description: g.Description,
+			Metadata:    g.Metadata,
+			ParentID:    g.ParentID,
+		}
+		return res, nil
+	}
+}
+
+func DecodeListGroupsRequest(_ context.Context, r *http.Request) (interface{}, error) {
```

Comment:
Consider separating decode methods from endpoint methods (break in two files `endpoint.go` and `transport.go`).